src/scripts/OSFramework/OSUI/Pattern/Video/Video.ts

```diff
@@ -0,0 +1,344 @@
+		// Store the platform events
+		private _platformEventOnStateChanged: Callbacks.OSOnStateChangedEvent;
+		// Store the video element
+		private _videoElement: HTMLVideoElement;
+		// Store the source element
+		private _videoSourceElement: HTMLSourceElement;
+		// Store the platform events
+
+		constructor(uniqueId: string, configs: JSON) {
+			super(uniqueId, new VideoConfig(configs));
+		}
+
+		// Method that will set the Autoplay attribute
+		private _setAutoplay(): void {
+			if (this.configs.Autoplay) {
+				this._videoElement.autoplay = true;
+
+				if (!this.configs.Muted) {
+					console.warn(
+						`Some modern browsers will not autoplay your video on ${GlobalEnum.PatternName.Video} if it's not muted. The general rule of many browsers is that a user must opt-in to certain actions before they can happen. Set the Muted parameter to True, to start the video automatically.`
+					);
+				}
+			} else {
+				this._videoElement.autoplay = false;
+			}
+		}
+
+		// Method that will set the controls attribute
+		private _setControls(): void {
+			if (this.configs.Controls) {
+				this._videoElement.controls = true;
+			} else {
+				this._videoElement.controls = false;
+			}
+		}
+
+		// Method that will set the Height attribute
+		private _setHeight(): void {
+			if (this.configs.Height !== Constants.EmptyString) {
+				OSUI.Helper.Dom.Attribute.Set(
+					this._videoElement,
+					Patterns.Video.Enum.VideoAttributes.Height,
+					this.configs.Height
+				);
+			} else {
+				OSUI.Helper.Dom.Attribute.Remove(this._videoElement, Patterns.Video.Enum.VideoAttributes.Height);
+			}
+		}
+
+		// Method that will set the loop attribute
+		private _setLoop(): void {
+			if (this.configs.Loop) {
+				this._videoElement.loop = true;
+			} else {
+				this._videoElement.loop = false;
+			}
+		}
+
+		// Method that will set the muted attribute
+		private _setMuted(): void {
+			if (this.configs.Muted) {
+				this._videoElement.muted = true;
+			} else {
+				this._videoElement.muted = false;
+			}
+		}
+
+		// Method that will set the poster image on video
+		private _setPosterUrl(): void {
+			if (this.configs.PosterURL !== Constants.EmptyString) {
+				this._videoElement.poster = this.configs.PosterURL;
+			} else {
+				this._videoElement.poster = Constants.EmptyString;
+			}
+		}
+
+		// Method to apply all initial video configs
+		private _setVideoConfigs(): void {
+			this._setAutoplay();
+
+			this._setControls();
+
+			this._setLoop();
+
+			this._setMuted();
+
+			this._setPosterUrl();
+
+			this._setWidth();
+
+			this._setHeight();
+
+			// Trigger the event of unstarted based on video progress
+			if (this._videoElement.currentTime === 0) {
+				this._triggerOnStateChangedEvent(Patterns.Video.Enum.VideoStates.Unstarted);
+			}
+		}
+
+		// Method create the source element
+		private _setVideoSource(): void {
+			// Get the file extension from URL
+			const urlFileExtension = this.configs.URL.split('.').pop();
```

Comment:
this is a private const, should be _urlFileExtension

Reply:
Done